addons/project_purchase/models/project_project.py

```diff
@@ -28,22 +28,22 @@ def _compute_purchase_orders_count(self):
     # ----------------------------
 
     def action_open_project_purchase_orders(self):
-        purchase_orders = self.env['purchase.order.line'].search(
-            [('analytic_distribution', 'in', self.analytic_account_id.ids)],
+        purchase_order_lines = self.env['purchase.order.line'].search(
+            [('analytic_distribution', 'in', self.analytic_account_id.ids)]
         )
```

Comment:
By doing that, you have less changes and it seems more relevant since following code doesn't care about the PO line, only purchase orders (you don't have to check `purchase_order_lines.order_id` every time)
```suggestion
        purchase_orders = self.env['purchase.order.line'].search(
            [('analytic_distribution', 'in', self.analytic_account_id.ids)]
        ).order_id
```